src/datasource.test.ts

```diff
@@ -1584,60 +1584,123 @@ describe('OpenSearchDatasource', function(this: any) {
-      it('should filter out ad hoc filter without key', () => {
-        const query = ctx.ds.addAdHocFilters('foo:"bar"', [{ key: '', operator: '=', value: 'a', condition: '' }]);
-        expect(query).toBe('foo:"bar"');
-      });
-      it('should filter out ad hoc filter without value', () => {
-        const query = ctx.ds.addAdHocFilters('foo:"bar"', [{ key: 'a', operator: '=', value: '', condition: '' }]);
-        expect(query).toBe('foo:"bar"');
+      describe('Lucene queries', () => {
+        it('should filter out ad hoc filter without key', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'foo:"bar"', queryType: QueryType.Lucene }, [
+            { key: '', operator: '=', value: 'a', condition: '' },
+          ]);
+          expect(query).toBe('foo:"bar"');
+        });
+
+        it('should filter out ad hoc filter without value', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'foo:"bar"', queryType: QueryType.Lucene }, [
+            { key: 'a', operator: '=', value: '', condition: '' },
+          ]);
+          expect(query).toBe('foo:"bar"');
+        });
+
+        it('should filter out filter ad hoc filter with invalid operator', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'foo:"bar"', queryType: QueryType.Lucene }, [
+            { key: 'a', operator: 'A', value: '', condition: '' },
+          ]);
+          expect(query).toBe('foo:"bar"');
+        });
       });
 
-      it('should filter out filter ad hoc filter with invalid operator', () => {
-        const query = ctx.ds.addAdHocFilters('foo:"bar"', [{ key: 'a', operator: 'A', value: '', condition: '' }]);
-        expect(query).toBe('foo:"bar"');
+      describe('PPL queries', () => {
+        it('should filter out ad hoc filter without key', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'source = test-index', queryType: QueryType.PPL }, [
+            { key: '', operator: '=', value: 'a', condition: '' },
+          ]);
+          expect(query).toBe('source = test-index');
+        });
+
+        it('should filter out ad hoc filter without value', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'source = test-index', queryType: QueryType.PPL }, [
+            { key: 'a', operator: '=', value: '', condition: '' },
+          ]);
+          expect(query).toBe('source = test-index');
+        });
+
+        it('should filter out filter ad hoc filter with invalid operators', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: 'source = test-index', queryType: QueryType.PPL }, [
+            { key: 'a', operator: '=~', value: '', condition: '' },
+            { key: 'a', operator: '!~', value: '', condition: '' },
+          ]);
+          expect(query).toBe('source = test-index');
+        });
       });
     });
 
     describe('with 1 ad hoc filter', () => {
```

Comment:
just curious - any reason why only testing Lucene in this test case?

Reply:
i didn't think there was anything that could be added under this test case with 1 ad hoc filter for PPL queries that couldn't be covered by the multiple ad hoc filters test cases.